crates/ruff/src/rules/perflint/rules/slow_filtered_list_creation.rs

```diff
@@ -0,0 +1,74 @@
+use rustpython_parser::ast;
+use rustpython_parser::ast::Expr;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::prelude::Stmt;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for cases where new lists are made by appending elements (with a filter) in a for-loop
+///
+/// ## Why is this bad?
+/// List comprehensions are 25% more efficient at creating new lists, with or without an
+/// if-statement. So these should be used when creating new lists
+///
+/// ## Example
+/// ```python
+/// original = range(10_000)
+/// filtered = []
+/// for i in original:
+///     if i % 2:
+///         filtered.append(i)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// original = range(10_000)
+/// filtered = [x for x in original if x % 2]
+/// ```
+#[violation]
+pub struct SlowFilteredListCreation;
+
+impl Violation for SlowFilteredListCreation {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use a list comprehension to create a new filtered list")
+    }
+}
+
+/// PERF401
+pub(crate) fn slow_filtered_list_creation(checker: &mut Checker, body: &[Stmt]) {
```

Comment:
@charliermarsh do we know whether it's cheaper to traverse the if-body inside of the rule OR call this check for every statement and test if the parent is an if statement?